proto/frequenz/api/common/v1/components.proto

```diff
@@ -10,6 +10,217 @@ syntax = "proto3";
 
 package frequenz.api.common.v1.components;
 
+import "frequenz/api/common/v1/metrics.proto";
+
+
+// ComponentData message aggregates multiple metrics, operational states, and
+// errors, related to a specific microgrid component.
+//
+// !!! example
+//     Example output structure:
+//     ```
+//     {
+//       "metric_samples": {
+//         "metric": "DC_VOLTAGE_V",
+//         "value": {
+//           "simple_metric": {
+//             "sampled_at": "2023-10-19T14:15:22.123456789Z",
+//             "value": 48.2,
+//             "bounds": [
+//               { "lower": 45.0, "upper": 50.0 }
+//             ]
+//           }
+//         },
+//         "metric": "DC_CURRENT_A": {
+//         "value": {
+//            "aggregated_metric": {
+//             "sampled_at": "2023-10-19T14:15:22.123456789Z",
+//             "avg_value": 10.2,
+//             "min_value": 9.8,
+//             "max_value": 10.6,
+//             "list_values": [9.8, 10.1, 10.6],
+//             "bounds": [
+//               { "lower": -20.0, "upper": -5.0 },
+//               { "lower": 5.0, "upper": 20.0 }
+//             ]
+//           }
+//         }
+//       },
+//       "state": "COMPONENT_STATE_CHARGING",
+//       "errors": []
+//     }
+//    ```
+message ComponentData {
+  // Representation of a sampled metric along with its value.
+  message MetricSample {
+    // The metric that was sampled.
+    frequenz.api.common.v1.metrics.Metric metric = 1;
+
+    // The value of the sampled metric.
+    frequenz.api.common.v1.metrics.MetricSampleVariant value = 2;
```

Comment:
And the prefix "frequenz.api.common.v1.metrics." can be remove as this is a self-reference.

Reply:
The prefix cannot be removed, because the definitions of `Metric` and `MetricSampleVariant` are in a different file (and therefore under a different package hierarchy).

Reply:
> I would suggest to call this sample and not value.

This depends on wether we keep the structure as suggest or change it to guarantee key uniqueness.

Reply:
Agree about the renaming. `sample` is more meaningful than `value` here.